delfin/drivers/fujitsu/eternus/eternus_stor.py

```diff
@@ -352,3 +391,548 @@ def parse_alert(context, alert):
     @staticmethod
     def get_access_url():
         return 'https://{ip}'
+
+    def list_storage_host_initiators(self, ctx):
+        initiator_list = []
+        host_status = self.get_host_status()
+        host_fc_list = self.get_data(consts.GET_HOST_WWN_NAMES,
+                                     consts.HOST_TOTAL)
+        for host_fc in (host_fc_list or []):
+            if len(host_fc) < consts.HOST_TOTAL:
+                continue
+            fc_name = host_fc[consts.HOST_NAME_COUNT]
+            fc_wwn = host_fc[consts.HOST_WWN_COUNT]
+            state = host_status.get(fc_name)
+            initiator_item = {
+                'name': fc_wwn,
+                'storage_id': self.storage_id,
+                'native_storage_host_initiator_id': fc_wwn,
+                'wwn': fc_wwn,
+                'status': constants.InitiatorStatus.ONLINE if
+                state is not None and state in
+                consts.HOST_PATH_STATUS_SPECIFIC_TWO else
+                constants.InitiatorStatus.OFFLINE,
+                'native_storage_host_id': fc_name,
+                'type': constants.InitiatorType.FC
+            }
+            initiator_list.append(initiator_item)
+        host_iscsi_list = self.get_iscsi_host()
+        for host_iscsi in (host_iscsi_list or []):
+            iscsi_name = host_iscsi.get('name')
+            state = host_status.get(iscsi_name)
+            initiator_item = {
+                "name": host_iscsi.get('iqn'),
+                "storage_id": self.storage_id,
+                "native_storage_host_initiator_id": host_iscsi.get('iqn'),
+                "wwn": host_iscsi.get('iqn'),
+                "status": constants.InitiatorStatus.ONLINE if
+                state is not None and state in
+                consts.HOST_PATH_STATUS_SPECIFIC_TWO else
+                constants.InitiatorStatus.OFFLINE,
+                "native_storage_host_id": iscsi_name,
+                'type': constants.InitiatorType.ISCSI,
+                'alias': host_iscsi.get('alias')
+            }
+            initiator_list.append(initiator_item)
+        host_sas_list = self.get_data(consts.GET_HOST_SAS_ADDRESSES,
+                                      consts.HOST_SAS_FIVE)
+        for host_sas in (host_sas_list or []):
+            if len(host_sas) < consts.HOST_SAS_FIVE:
+                continue
+            sas_name = host_sas[consts.HOST_SAS_ONE]
+            sas_address = host_sas[consts.HOST_SAS_TWO]
+            state = host_status.get(sas_name)
+            initiator_item = {
+                "name": sas_address,
+                "storage_id": self.storage_id,
+                "native_storage_host_initiator_id": sas_address,
+                "wwn": sas_address,
+                "status": constants.InitiatorStatus.ONLINE if
```

Comment:
consts.HOST_PATH_STATUS_SPECIFIC_TWO is a specific status value, so use = = instead of in for judgment.
![image](https://user-images.githubusercontent.com/59107077/153788992-408f0263-0214-45cd-b9c4-84e487446486.png)

Reply:
Has been modified